src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -1512,8 +1512,161 @@ internal override void ForceComplete(SourceLocation locationOpt, Predicate<Symbo
+                            }
+                        }
+                        break;
+
+                    case CollectionExpressionTypeKind.CollectionBuilder:
+                        {
+                            var syntax = ParameterSyntax;
+                            var binder = GetDefaultParameterValueBinder(syntax); // this binder is good for our purpose
+
+                            binder.TryGetCollectionIterationType(syntax, Type, out elementTypeWithAnnotations);
+                            elementType = elementTypeWithAnnotations.Type;
+                            if (elementType is null)
+                            {
+                                reportInvalidParams(diagnostics);
+                                return;
+                            }
+
+                            MethodSymbol? collectionBuilderMethod = binder.GetAndValidateCollectionBuilderMethod(syntax, (NamedTypeSymbol)Type, diagnostics, elementType: out _);
+                            if (collectionBuilderMethod is null)
+                            {
+                                return;
+                            }
+
+                            if (ContainingSymbol.ContainingSymbol is NamedTypeSymbol) // No need to check for lambdas or local function
+                            {
+                                checkIsAtLeastAsVisible(syntax, binder, collectionBuilderMethod, diagnostics);
+                            }
+                        }
+                        break;
+                }
+
+                Debug.Assert(elementType is { });
+
+                if (collectionTypeKind != CollectionExpressionTypeKind.Array)
+                {
+                    MessageID.IDS_ParamsCollections.CheckFeatureAvailability(diagnostics, ParameterSyntax);
+                }
+
+                bool isAtLeastAsVisible(ParameterSyntax syntax, Binder binder, MethodSymbol method, BindingDiagnosticBag diagnostics)
+                {
+                    var useSiteInfo = binder.GetNewCompoundUseSiteInfo(diagnostics);
+
+                    bool result = method.IsAsRestrictive(ContainingSymbol, ref useSiteInfo) &&
+                                  method.ContainingType.IsAtLeastAsVisibleAs(ContainingSymbol, ref useSiteInfo);
+
+                    diagnostics.Add(syntax.Location, useSiteInfo);
+                    return result;
+                }
+
+                void checkIsAtLeastAsVisible(ParameterSyntax syntax, Binder binder, MethodSymbol method, BindingDiagnosticBag diagnostics)
+                {
+                    if (!isAtLeastAsVisible(syntax, binder, method, diagnostics))
+                    {
+                        diagnostics.Add(ErrorCode.ERR_ParamsMemberCannotBeLessVisibleThanDeclaringMember, syntax, method, ContainingSymbol);
+                    }
+                }
+            }
+
+            void reportInvalidParams(BindingDiagnosticBag diagnostics)
```

Comment:
>reportInvalidParams [](http://example.com/codeflow?start=17&length=19)

Consider declaring this local function and the two above at the same scope.

Reply:
> Consider declaring this local function and the two above at the same scope.

Done